app/routers/analyze.py

```
@@ -0,0 +1,67 @@
+
+    # Extract text
+    parser = ParserFactory.get_parser(file_extension)
+    extracted_text = parser.extract_text(temp_file_path)
+
+    # Cleanup temp file
+    os.remove(temp_file_path)
+
+    return {"extracted_text": extracted_text}
+
+@router.post("/analyze/")
+async def analyze_document(file: UploadFile = File(...)):
```

Ruff (0.8.2)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)